### PR TITLE
fix profile sales channel and zipCode

### PR DESCRIPTION
**Pull request by RipMace**

### CODE REVIEW

File: src/components/Form/CreateProfileForm/ProfileData/ProfileData.tsx [hunk 1]
```diff
@@ -59,20 +60,36 @@ const ProfileData = ({
...
     if (agreement) {
-      void Api.Profile.updateProfile(agreement.id, discount).then(() =>
-        onUpdate()
-      );
+      void Api.Profile.updateProfile(agreement.id, discount)
+        .then(() => {
+          onUpdate();
+          handleNext();
+        })
+        .catch(throwErrorTooltip);
```
Comment (AleDore) [inline src/components/Form/CreateProfileForm/ProfileData/ProfileData.tsx:92]: Prefer use `taskEither` by wrapping Promise with `tryCatch` :)

File: src/components/Form/CreateProfileForm/ProfileData/ProfileData.tsx [hunk 2]
```diff
@@ -59,20 +60,36 @@ const ProfileData = ({
...
+  };
+
   const createProfile = (discount: any) => {
     if (agreement) {
-      void Api.Profile.createProfile(agreement.id, discount);
+      void Api.Profile.createProfile(agreement.id, discount)
+        .then(() => {
+          handleNext();
+        })
+        .catch(throwErrorTooltip);
```
Comment (AleDore) [inline src/components/Form/CreateProfileForm/ProfileData/ProfileData.tsx:81]: ditto :)

File: src/components/Form/CreateProfileForm/ProfileData/ProfileData.tsx (hunk 1 above)
Comment (RipMace) [reply src/components/Form/CreateProfileForm/ProfileData/ProfileData.tsx:92]: ` await tryCatch(
      () => Api.Profile.updateProfile(agreement.id, discount),
      toError
    )
      .fold(throwErrorTooltip, () => {
        onUpdate();
        handleNext();
      })
      .run();`
      
      This never goes on Left when the API fail, I'm missing something?

**Review by AleDore — APPROVED** [review]
LGTM, maybe we have to propagate this fix to all others api calls :)

**Pull request merged**